spec/src/antlr/JDQL.g4

```diff
@@ -0,0 +1,92 @@
+grammar JDQL;
+
+statement : select_statement | update_statement | delete_statement;
+
+select_statement : select_clause? from_clause? where_clause? orderby_clause?;
+update_statement : 'UPDATE' entity_name set_clause where_clause?;
+delete_statement : 'DELETE' from_clause where_clause?;
+
+from_clause : 'FROM' entity_name;
+
+where_clause : 'WHERE' conditional_expression;
+
+set_clause : 'SET' update_item (',' update_item)*;
+update_item : state_field_path_expression '=' (scalar_expression | 'NULL');
+
+select_clause : 'SELECT' select_list;
+select_list
+    : scalar_expression (',' scalar_expression)* // could limit to single select item and/or to state_field_path_expression
+    | aggregate_expression
+    ;
+aggregate_expression : 'COUNT' '(' '*' ')';
+
+orderby_clause : 'ORDER' 'BY' orderby_item (',' orderby_item)*;
+orderby_item : state_field_path_expression ('ASC' | 'DESC');
+
+conditional_expression
+    // highest to lowest precedence
+    : '(' conditional_expression ')'
+    | null_comparison_expression
+    | in_expression
+    | between_expression
+    | like_expression
```

Comment:
perhaps a statement that regex could be supported in a like expression?

Reply:
So this is a question I was asking myself. From what little I have seen, very many non-relational databases seem to have good support for regexes. _Traditionally_, relational databases did not, though I guess most do by now. The JPQL `like` operator is the ANSI SQL one. If we wanted to provide support for regexes, we would need to provide some syntax for that. The `REGEXP_LIKE` function seems to be the most standard thing in SQL, though I would say it's pretty ugly. (I forget what the ANSI SQL standard for this is.)

Reply:
Ah, right, so ANSI SQL has a `SIMILAR TO` operator, and its own dialect of regular expressions.

Reply:
We discussed this on a recent JD call, and there was little enthusiasm for adding regex support _in this first release_.

But @graemerocher, if this is important to you, we can rediscuss it. Perhaps you could open an issue, or propose a PR.

We don't need to do it in _this_ PR.